src/components/Modals/AccessDeniedNoAuthModal.js

```diff
@@ -1,46 +1,42 @@
 import React from "react";
-import Modal from 'react-modal';
 import { useDispatch, useSelector } from "react-redux";
 import { useTranslation } from "react-i18next";
 
 import Button from "../Button/Button";
 import '../../Modal.scss';
 import { closeAccessDeniedNoAuthModal } from "../Editor/EditorSlice";
-import { CloseIcon } from "../../Icons";
 import LoginButton from "../Login/LoginButton";
+import GeneralModal from "./GeneralModal";
+import { login } from "../../utils/login";
 
 const AccessDeniedNoAuthModal = () => {
   const dispatch = useDispatch()
   const { t } = useTranslation();
   
   const isModalOpen = useSelector((state) => state.editor.accessDeniedNoAuthModalShowing)
+  const accessDeniedData = useSelector((state) => state.editor.modals.accessDenied)
   const closeModal = () => dispatch(closeAccessDeniedNoAuthModal());
 
+  const defaultCallback = () => {
+    login({accessDeniedData})
```

Comment:
🤔 I know the [src/utils/login.js](https://github.com/RaspberryPiFoundation/editor-ui/pull/491/files#diff-f6d1c1aacbdfc7531661752e1ddbd8d77447464bb092eb75e614a989c93e1117) was a copy paste but the if/else in there and passing in this state variable makes me a little nervous. Think it's just a broader question around state values but it feels like this could easily error. One for another PR but wanted to highlight it